api/leetcode-data.js

```diff
@@ -0,0 +1,104 @@
+import { dailyGetLeetcodeData, weeklyGetLeetcodeData, getProblemSet } from "./leetcode/index.js";
+import logger from "../logging.js";
+import staticProblemSets from './leetcode/static/problemSets.json' assert { type: "json" };
+
+export class LeetcodeData {
+
+    constructor() {}
+
+    getProblemSetData = async (limit, skip) => {
+        const { data: { problemsetQuestionList: { questions } } } = await getProblemSet(limit, skip);
+        if (!questions) {
+            logger.error("Unable to fetch problemsetQuestionList");
+            return;
+        }
+        return questions
+    };
+
+    getNumProblems = async () => {
+        const { data: { problemsetQuestionList: { total } } } = await getProblemSet();
+        if (!total) {
```

Comment:
ditto, but up to you is you want to live with this instead becase is clearer for you.